src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -34,6 +34,54 @@ void FilterRemovedNodes(std::shared_ptr<gcs::GcsClient> gcs_client,
   }
 }
 
+/// Update object location data based on response from the owning core worker.
+bool UpdateObjectLocations(const rpc::GetObjectLocationsOwnerReply &location_reply,
+                           const Status &status, const ObjectID &object_id,
+                           std::shared_ptr<gcs::GcsClient> gcs_client,
+                           std::unordered_set<NodeID> *node_ids, std::string *spilled_url,
+                           NodeID *spilled_node_id, size_t *object_size) {
+  bool is_updated = false;
+
+  std::unordered_set<NodeID> new_node_ids;
+
+  if (!status.ok()) {
+    RAY_LOG(INFO) << "Failed to return location updates to subscribers  for " << object_id
+                  << ": " << status.ToString()
+                  << ", assuming that the object was freed or evicted.";
+    *node_ids = new_node_ids;
```

Comment:
Before this refactoring, this block sets *object_size = 0. It is not necessary?

Reply:
I don't think so, no. My original thinking was that we'd want to consider the object size to be 0 for an object that has been freed/evicted in order to free up pull bandwidth, but it doesn't look like that's how the pull manager works. And AFAICT, the current GCS-based object directory doesn't reset the object size to 0 upon the object being freed/evicted.